roles/app/tasks/main.yml

```diff
@@ -38,14 +38,33 @@
     - "{{ datadir }}/media"
     - "{{ datadir }}/static"
 
+- name: clone the application
+  become: true
+  become_user: "{{ appuser }}"
+  git:
+    repo: "{{ apprepo }}"
+    dest: "{{ appdir }}"
+    update: no
+  
+
+- name: add git remotes
```

Comment:
I don't really like this approach because it runs the commands every time, and voids any manual modifications to the `remote.origin.fetch` config param every time.

Maybe something like setting a variable with the refspec ; `+refs/pull/*:refs/pull/*` if the version matches `pull/.*`, `+refs/heads/*:refs/remotes/origin/*` if not.